patch/1.25.3.1/lua-resty-core-shared_shdict.patch

```diff
@@ -0,0 +1,276 @@
+diff --git lib/resty/core/shdict.lua lib/resty/core/shdict.lua
+index dedf12c..7644b77 100644
+--- lib/resty/core/shdict.lua
++++ lib/resty/core/shdict.lua
+@@ -28,7 +28,6 @@ local type = type
+ local error = error
+ local getmetatable = getmetatable
+ local FFI_DECLINED = base.FFI_DECLINED
+-local subsystem = ngx.config.subsystem
+ 
+ 
+ local ngx_lua_ffi_shdict_get
+@@ -42,117 +41,175 @@ local ngx_lua_ffi_shdict_free_space
+ local ngx_lua_ffi_shdict_udata_to_zone
+ 
+ 
+-if subsystem == 'http' then
+-    ffi.cdef[[
+-int ngx_http_lua_ffi_shdict_get(void *zone, const unsigned char *key,
++ffi.cdef[[
++int ngx_meta_lua_ffi_shdict_get(void *zone, const unsigned char *key,
+     size_t key_len, int *value_type, unsigned char **str_value_buf,
+     size_t *str_value_len, double *num_value, int *user_flags,
+     int get_stale, int *is_stale, char **errmsg);
+ 
+-int ngx_http_lua_ffi_shdict_incr(void *zone, const unsigned char *key,
++int ngx_meta_lua_ffi_shdict_incr(void *zone, const unsigned char *key,
+     size_t key_len, double *value, char **err, int has_init,
+     double init, long init_ttl, int *forcible);
+ 
+-int ngx_http_lua_ffi_shdict_store(void *zone, int op,
++int ngx_meta_lua_ffi_shdict_store(void *zone, int op,
+     const unsigned char *key, size_t key_len, int value_type,
+     const unsigned char *str_value_buf, size_t str_value_len,
+     double num_value, long exptime, int user_flags, char **errmsg,
+     int *forcible);
+ 
+-int ngx_http_lua_ffi_shdict_flush_all(void *zone);
++int ngx_meta_lua_ffi_shdict_flush_all(void *zone);
+ 
+-long ngx_http_lua_ffi_shdict_get_ttl(void *zone,
++long ngx_meta_lua_ffi_shdict_get_ttl(void *zone,
+     const unsigned char *key, size_t key_len);
+ 
+-int ngx_http_lua_ffi_shdict_set_expire(void *zone,
++int ngx_meta_lua_ffi_shdict_set_expire(void *zone,
+     const unsigned char *key, size_t key_len, long exptime);
+ 
+-size_t ngx_http_lua_ffi_shdict_capacity(void *zone);
+-
```

Comment:
The implementation has been change in ngx_lua-0.1.28. So the patch will fail to apply